### PR TITLE
add integration and unit tests for esm builds

### PR DESCRIPTION
**Pull request by erwinmombay**
in preparation for module/nomodule launch we need to start running the module build with its own tests runs

### CODE REVIEW

File: build-system/compile/build.conf.js [hunk 1]
```diff
@@ -121,6 +121,7 @@ function getJsonConfigurationPlugin() {
  * @return {!Array<string|!Array<string|!Object>>}
  */
 function plugins({isEsmBuild, isForTesting, isSinglePass, isChecktypes}) {
+  console.log('isEsmBuild', isEsmBuild);
```
Comment (rsimha) [inline build-system/compile/build.conf.js]: Remember to delete this!

File: build-system/pr-check/esm-tests.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview
+ * This script builds and tests the AMP runtime in single pass mode.
```
Comment (rsimha) [inline build-system/pr-check/esm-tests.js]: Fix comment ("single pass mode" -> "esm mode")

File: build-system/pr-check/esm-tests.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+ * This is run during the CI stage = test; job = single pass tests.
```
Comment (rsimha) [inline build-system/pr-check/esm-tests.js]: Fix comment: "job = esm tests"

File: build-system/pr-check/esm-tests.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+const {
+  printChangeSummary,
+  startTimer,
+  stopTimer,
+  timedExecOrDie: timedExecOrDieBase,
+} = require('./utils');
+const {determineBuildTargets} = require('./build-targets');
+const {isTravisPullRequestBuild} = require('../common/travis');
+
+const FILENAME = 'single-pass-tests.js';
```
Comment (rsimha) [inline build-system/pr-check/esm-tests.js]: `esm-tests.js`

File: build-system/pr-check/esm-tests.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    ) {
+      timedExecOrDie('gulp update-packages');
+      timedExecOrDie('gulp dist --fortesting --esm');
+      timedExecOrDie(
+        'gulp integration --nobuild --compiled --headless'
+      );
+    } else {
+      console.log(
+        `${FILELOGPREFIX} Skipping`,
+        colors.cyan('ESM Pass Tests'),
```
Comment (rsimha) [inline build-system/pr-check/esm-tests.js]: Fix name: "ESM Pass Tests" -> "ESM Tests"

**Comment by erwinmombay** [comment]
obsolete and superseded

**Pull request closed without merging**